oeps/processes/oep-0056-architectural-advisory-process.rst

```diff
@@ -0,0 +1,197 @@
+
+Atrophy of architectural principles and tech radar
+--------------------------------------------------
+
+An important component of our ability to make architectural decisions that are aligned across teams and organizations is measuring and vetting those decisions against a set of `shared architectural principles`_ and technologies in our `Tech Radar`_.  Today, we lack organized maintenance of our principles and radar, nor do we have an appropriate forum in which to discuss and socialize them.
+
+Specification
+=============
+
+We believe that we can create a healthy and empowering dialogue around our architectural process and mitigate the above concerns by clarifying the spirit and process around the creation and review of ADRs.
+
+We will adjust our current architectural process by:
+
+- Formalizing and clarifying an **Architectural Advisory Process** in which we conversationally provide input on and promptly review the status of ADRs, as well as review updates to our architectural principles and tech radar, with the goal of enabling more effective, informed decision making.
+- Improve the discoverability of stakeholders and experts via a **decision maker's checklist**.
+- Set decision makers up for success by using the Architectural Advisory Process to **vet changes to our architectural principles and tech radar**.  Both are crucial inputs into the decision making process by helping to align our community on our architectural values and technology choices.
+
+Architectural Advisory Process (AAP)
+------------------------------------
+
+We'll create an Architecture Advisory Process to add structure to today's informal review of ADRs.  The goal of this process is to review ADRs with relevant stakeholders and experts.  Conversations in the advisory process need to take place promptly so that decisions do not feel blocked on gathering advice.
+
+We intend to also use the Advisory Process over time to revisit and iterate on our architectural principles with the goal of ensuring they're specific, measurable, achievable, realistic, and testable. (a variation of SMART criteria)  Similarly, like ADRs and architectural principles, we can use the process to review and adjust Open edX Tech Radar blips.
+
+This process is intended to be lightweight, approachable, and asynchronous.  See :ref:`principles` for more details on the spirit and functioning of the Advisory Process.
+
+Advisory Process Artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The final artifact of the Advisory Process is an Architectural Decision Record document checked in to a repository relevant to the decision.  Prior to that, however, we suggest as a best practice using Github's Issues and Pull Request tooling to conduct the Advisory Process.  The advantage of doing so is that Issue/Pull Request comments remove the burden of recording who gave particular advice, and allow us to have threaded, focused conversations about various aspects of ADR.
```

Comment:
Two comments:
1. Here you describe how to implement recording advice, before detailing it as a requirement, which doesn't come until the appendix. I don't see why the appendix is any less important the rest of this OEP for setting up context, and wish it came before this section.
```suggestion
The final artifact of the Advisory Process is an Architectural Decision Record document checked in to a repository relevant to the decision.  Prior to that, however, we suggest as a best practice using Github's Issues and Pull Request tooling to conduct the Advisory Process.  The advantage of doing so is that Issue/Pull Request comments remove the burden of recording who gave particular advice, and allow us to have threaded, focused conversations about various aspects of an ADR.
```

Reply:
Regarding the first point - this has been a struggle.  I've received feedback both that the stuff in the Appendix is long and should have been broken out like it is, and also that it's necessary context.  I originally just had it all inline with no appendix.  Maybe this indicates I should find a different way to present this?

Reply:
Here's a variety of thoughts around the Appendix that you can take or leave.
1. To me "Appendix" means "optional reading". I don't believe all parts are equally optional.
2. You may consider listing the principles (through internal TOC or repetition with short description), with the Appendix to learn more.
3. If important text is too long, consider bullets or removing/moving extra explanatory text.
4. The optional GitHub Issue section is another example of lengthy optional text in the doc. It could instead be a brief note with a link to more.

Note: none of this materially changes the proposal. However, since there will be many more readers than writers, it may help to give this some attention.

Thanks, whatever you choose to do.

Reply:
Thanks @robrap, I like these ideas.  I'm going to mix it around a bit and see if I can find something that feels better.